File: src/main/java/vergisst/minecraftmod/weaponthrow/packets/CPacketThrow.java
package vergisst.minecraftmod.weaponthrow.packets;

import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking;
import vergisst.minecraftmod.weaponthrow.handler.EventsHandler;

public class CPacketThrow extends BasePacket {
    public CPacketThrow(State state) {
        super(PacketIdentifiers.CPACKET_THROW);
        buf.writeByte(state.toByte());
    }

    public static void register() {
        ServerPlayNetworking.registerGlobalReceiver(PacketIdentifiers.CPACKET_THROW, (server, player, handler, buf, responseSender) -> {
            State action = State.fromByte(buf.readByte());

            server.execute(() -> {
                EventsHandler.onThrowItem(player, action);
            });
        });
    }
}

File: src/main/kotlin/vergisst/minecraftmod/weaponthrowlite/packets/CPacketThrow.kt
package vergisst.minecraftmod.weaponthrowlite.packets

import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking
import vergisst.minecraftmod.weaponthrowlite.handlers.EventsHandler
import vergisst.minecraftmod.weaponthrowlite.handlers.PacketsIdentifier

class CPacketThrow(val state: PacketState): BasePacket(PacketsIdentifier.CPACKET_THROW) {
    init {
        buf?.writeByte(state.toByte().toInt())
    }

    companion object {
        fun register() {
            ServerPlayNetworking.registerGlobalReceiver(
                PacketsIdentifier.CPACKET_THROW,
                { server, player, handler, buf, responseSender ->
                    val action = PacketState.fromByte(buf.readByte().toInt())

                    server.execute {
                        EventsHandler.onThrowItem(player, action)
                    }
                }
            )
        }
    }
}